### ISSUE TITLE
Menu manager deletes third level

### ISSUE THREAD

**Issue by syntaxerror** [arastta/arastta#392]
To test:
1. Make two "top" menu items.
2. In the second add at least one sub-level, so you get a two level structure.
3. Refresh the site or look at front-end, all should be fine.
4. Drag and drop the two level item into the other, so you get a three level structure.
5. Refresh the site or look at front-end, and the third level will be deleted.

Repeat step 1-4
5. Oh, that was a mistake, so we drag it back to where it was.
6. Refresh or look at front-end, second level is missing, because it was deleted when it was a third level.

Clearly buggy as it is, but I think the menu manager should support more than two levels anyway ...

**Comment by syntaxerror** [comment]
Is there any reason for menus (top and side module) being limited to two levels?
Quite many stores needs more than that ... ;)

**Comment by gepeixoto** [comment]
I agree with Rune. I was trying to change menu.js in order to block items with children to be added as other's child due to this bug - without success =]

**Comment by gepeixoto** [comment]
In admin/view/javascript/menu/menu.js, I found that the function updateSharedVars(ui) have a condition that controls maxDepth menu level:
`if( prev.length )`
`maxDepth = ( (depth = prev.menuItemDepth() + 1) > api.options.globalMaxDepth ) ? api.options.globalMaxDepth : depth;`
`else maxDepth = 0;`
We can prevent parent menus that have children to be added as other's child if we could check this:
`if( prev.length && !this.haveChildrenMenus)`
But I'm not finding a way to check for children menus in current item.

**Comment by gepeixoto** [comment]
Ok, I found. By adding **!transport.children()[0]** to the condition:
`if( prev.length && !transport.children()[0] )`
This way, if the menu being dragged have children menus, it will be unable to be added as other's child, only as a parent menu, preventing this bug.
At least, a workaround until 3rd level menu support.
